me.sergiotarxz.JapaChar.yml

```
@@ -0,0 +1,99 @@
+        - '-des'
+        # Build a shared library.
+        - '-Duseshrplib'
+    build-options:
+        cflags: '-fPIC'
+        ldflags: '-fpic'
+    sources:
+        - type: archive
+          url: https://www.cpan.org/src/5.0/perl-5.40.0.tar.gz
+          sha256: c740348f357396327a9795d3e8323bafd0fe8a5c7835fc1cbaba0cc8dfe7161f
+        - type: shell
+          commands:
+            # Have Flatpak run the GNU-compatible configure script.
+            - 'ln -s configure{.gnu,}'
+      # Restore write permission to the Perl libraries.
+    post-install:
+        - 'chmod -R u+w /app/lib/perl5'
+      # Clean up a bunch of stuff we don't need. Depending on your application,
+      # you may have to drop some of these (e.g. *.pod).
+    cleanup:
+        - '/bin/corelist'
+        - '/bin/cpan'
+        - '/bin/enc2xs'
+        - '/bin/encguess'
+        - '/bin/h2ph'
+        - '/bin/h2xs'
+        - '/bin/instmodsh'
+        - '/bin/json_pp'
+        - '/bin/libnetcfg'
+        - '/bin/perlbug'
+        - '/bin/perldoc'
+        - '/bin/perlivp'
+        - '/bin/perlthanks'
+        - '/bin/piconv'
+        - '/bin/pl2pm'
+        - '/bin/pod*'
+        - '/bin/prove'
+        - '/bin/ptar*'
+        - '/bin/shasum'
+        - '/bin/splain'
+        - '/bin/xsubpp'
+        - '/bin/zipdetails'
+        - '/include'
+        - '/man'
+        - '*.pod'
+
+  # Installs the modules generated by flatpak-cpan-generator.
+  # Note that *any* Makefile.PL-style modules MUST be installed in this one step,
+  # as once perllocal.pod is written in one module, it cannot be modified by others.
+  - name: perl-libs
+    buildsystem: simple
+    build-commands:
+      - 'perl-libs/install.sh'
+    # Same as with the Perl module, we need to restore write permission.
+    # However, -f is now passed to avoid errors from trying to touch files from the
+    # above module that are now marked as read-only.
+    post-install:
+      - 'chmod -Rf u+w /app/lib/perl5/site_perl'
+    sources:
+      - generated-sources.json
+    # This step should be customized based on the CPAN packages you're using.
+    cleanup:
+      - '/bin'
+      - '/man'
+  - name: "japachar"
+    buildsystem: simple
+    build-commands:
+          - 'japachar/install.sh'
+    sources: 
+      - type: "git"
+        url: 'https://git.owlcode.tech/sergiotarxz/JapaChar'
+        tag: "v0.01"
+        dest: 'JapaCharSource'
+      - commands: 
+          - 'set -e'
+          - 'cp -vr JapaCharSource ${FLATPAK_DEST}/JapaChar'
+          - 'mkdir -pv /app/share/applications'
+          - 'cp -v JapaCharSource/me.sergiotarxz.JapaChar.desktop /app/share/applications/'
+          - 'cp -v JapaCharSource/run.sh /app/bin/japachar'
+          - 'mkdir -pv /app/share/icons/hicolor/scalable/apps/'
+          - 'cp -pv JapaCharSource/me.sergiotarxz.JapaChar.svg /app/share/icons/hicolor/scalable/apps/'
+        dest: "japachar"
+        dest-filename: "install.sh"
+        type: "script"
```

```suggestion
    build-commands:
          - 'cp -vr JapaCharSource ${FLATPAK_DEST}/JapaChar'
          - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.desktop -t /app/share/applications/'
          - 'install -Dm755 JapaCharSource/run.sh /app/bin/japachar'
          - 'install -Dm644 JapaCharSource/me.sergiotarxz.JapaChar.svg -t /app/share/icons/hicolor/scalable/apps/'
    sources: 
      - type: "git"
        url: 'https://git.owlcode.tech/sergiotarxz/JapaChar'
        tag: "v0.01"
        dest: 'JapaCharSource'
```

no need to make things complicated.

---

Fixed